test/test_recipes_integration.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+import os
+import subprocess
+import json
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def recipe_name(recipe):
+    return recipe['recipe'].strip()
+
+
+def should_skip():
+    # Skip if not in cron job task
+    return os.getenv('TRAVIS_EVENT_TYPE') != 'cron'
+
+
+def fail(recipe_integration, message):
+    if recipe_integration.get('xfail'):
+        pytest.xfail(message)
+    else:
+        pytest.fail(msg=message)
+
+
+def test_recipe_integration(recipe_integration):
+    if should_skip():
+        pytest.skip("Not run by cron job")
+
+    error = None
+    try:
+        command = "adr --format json {}".format(recipe_integration.get('recipe')).strip()
+        data = subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT)
+        result = json.loads(data)
+        if not result or len(result) == 0:
+            raise ValueError("Invalid data {}".format(data))
+    except Exception as e:
+        error = str(e)
+
+    if error:
+        fail(recipe_integration, error)
```

Comment:
Might as well put this directly in the `except` and get rid of the error variable.

Reply:
Ok I did it